src/Features/Core/Portable/DocumentIdSpan.cs

```diff
@@ -13,24 +14,27 @@ namespace Microsoft.CodeAnalysis
     /// pointing at a particular location in a <see cref="Document"/> but do not want to root a potentially
     /// very stale <see cref="Solution"/> snapshot that may keep around a lot of memory in a host.
     /// </summary>
+    [DataContract]
     internal readonly struct DocumentIdSpan
     {
-        private readonly Workspace _workspace;
```

Comment:
remove workspace.  That allowed thsi type to become serializable, which made it so all the types in inheritance margin can be serializable themselves (no need for a duplicate hierarchy).